src/functions/assertions/PesterThrow.ps1

```diff
@@ -124,13 +124,21 @@
         $but = Join-And $buts
         $failureMessage = "Expected an exception$(if($filter) { " with $filter" }) to be thrown,$(Format-Because $Because) but $but. $actualExceptionLine".Trim()
 
-        return [PSCustomObject] @{
+    $ActualValue = $actualExceptionMessage
+    $ExpectedValue = if ($filterOnExceptionType) { "type $(Format-Nicely $ExceptionType)" } else { 'any exception' }
+
+        return [Pester.ShouldResult] @{
```

Comment:
Will people be still able to register their own assertions without using our types? I think so, but are there tests for it?

Reply:
As long as they don't provide a object with properties/keys other than `FailureMessage` and `Succeeded`, including `ExpectResult` with wrong type/props. That would cause InvalidCastExpection

Reply:
`FailureMessage` and `Succeeded` are documented as [required properties in custom assertions](https://pester.dev/docs/assertions/custom-assertions#requirements--interface). As this only affects `Should`-registered operators, is documented and any additional properties/keys are inaccessible for the user, I don't think this is breaking in any way - even though validation will be stricter.

> I think so, but are there tests for it?

I don't think we have tests actually invoking custom assertions atm. Only `Add-ShouldOperator.ts.ps1` that makes sure they get registered properly. Should definitely add some P-tests.